libcst/matchers/_matcher_base.py

```diff
@@ -133,7 +145,16 @@ def __init__(self, *options: Union[_MatcherT, "OneOf[_MatcherT]"]) -> None:
 
         if len(actual_options) < 2:
             raise Exception("Must provide at least two options to OneOf!")
-        self.options: Sequence[_MatcherT] = actual_options
+        self._options: Sequence[_MatcherT] = tuple(actual_options)
+
+    @property
+    def options(self) -> Sequence[_MatcherT]:
```

Comment:
This is fine with me, but tbh it's not really needed for this diff; if we hash only by identity, not by value, it no longer matters whether the instance attributes are mutable; the only thing that must be immutable is the identity, which is by definition immutable.

Reply:
I still like the idea of providing an Exception when somebody tries to update a constructed matcher. We already do that for the generated classes, so I'll elect to keep this for special matchers to make sure there's no gap.